### DISCUSSION TITLE
Provide own AccessToken (not RefreshToken)

### DISCUSSION THREAD

**Discussion by tank104** [abuzuhri/Amazon-SP-API-CSharp#483] [Ideas]
Is it possible to provide our own AccessToken instead of RefreshToken. 
We have our own OAuth security microservice which handles all this, and would like to now give access to RefreshTokens for security reasons (since MS can be consumed externally).

Is it currently possible?  If not I am happy to do a PR for it - just wondered if you would be opposed to the idea?

**Comment by abuzuhri** [comment]
Its work for most of with refresh token what the problem with it ?

**Comment by tank104** [comment]
There is no problem with Refresh Token - the problem is that we try and not share around Refresh Tokens in our application microservices (since it is long lasting), we currently just get AccessTokens for all our marketplace integrations from our Auth service (since if they got in to wrong hands they don't last long).

So our preference is to give the package the AccessToken (instead of you generating it). 

Can we add this into the package?  (I am happy to do it, but if there are any pointers that would help be much appreciated)
